### PR TITLE
Use `force_iter_type` in broadcast cases

### PR DESCRIPTION
**Pull request by Priya2698**
Fixes Issue #2532.

### CODE REVIEW

**Comment by Priya2698** [comment]
!build

File: csrc/ops/utils.cpp [hunk 1]
```diff
@@ -350,11 +350,14 @@ IterDomain* newOutputIterDomain(
             .iter_type(iter_type.value())
             .build();
   } else {
+    if (!iter_type.has_value()){
+      iter_type = IterType::Broadcast;
+    }
```
Comment (jacobhinkle) [inline csrc/ops/utils.cpp:355]: Can you say more about when we might hit a case where iter_type is not null or broadcast?

Comment (Priya2698) [reply csrc/ops/utils.cpp:355]: Will add a comment.
Right now, I am checking if this fix works generally, and it looks like we cannot have reduction axis with extent 1. Will look at alternative solutions.

**Comment by Priya2698** [comment]
Closing this, as reduction axis cannot have extent 1.

**Pull request closed without merging**